### PR TITLE
Make ProcessUtility.Execute sync

### PR DESCRIPTION
**Pull request by yufeih**
#4047 #4259

### CODE REVIEW

File: src/docfx/lib/git/GitUtility.cs [hunk 1]
```diff
@@ -314,23 +311,23 @@ private static async Task CloneOrUpdate(string path, string url, IEnumerable<str
 
             try
             {
-                await ExecuteNonQuery(path, $"{httpConfig} fetch --tags --progress --update-head-ok {pruneSwitch} {depth} \"{url}\" {refspecs}", stderr: true);
+                ExecuteNonQuery(path, $"{httpConfig} fetch --tags --progress --update-head-ok {pruneSwitch} {depth} \"{url}\" {refspecs}");
             }
-            catch (InvalidOperationException ex) when (committishes.Any(rev => ex.Message.Contains(rev)))
+            catch (InvalidOperationException)
```
Comment (yufeih) [inline src/docfx/lib/git/GitUtility.cs]: removed `stderr` from `ProcessUtility.Execute`, we'll retry here in most cases

File: src/docfx/lib/log/DocfxException.cs [hunk 2]
```diff
@@ -21,8 +21,9 @@ public DocfxException(Error error, Exception innerException = null)
 
         public static bool IsDocfxException(Exception ex, out DocfxException docfxException)
         {
-            docfxException = ex as DocfxException ?? ex.InnerException as DocfxException;
-            return docfxException != null;
+            docfxException = ex as DocfxException;
+
+            return docfxException != null ? true : IsDocfxException(ex.InnerException, out docfxException);
         }
```
Comment (yufeih) [inline src/docfx/lib/log/DocfxException.cs]: This is because of nested `AggregatedException`

File: src/docfx/lib/ProcessUtility.cs [hunk 3]
```diff
@@ -181,81 +179,30 @@ public static bool ReleaseExclusiveLock(string lockName, string acquirer)
...
-                }
-                else
-                {
-                    var message = $"'\"{fileName}\" {commandLineArgs}' failed in directory '{cwd}' with exit code {process.ExitCode}: \nSTDOUT:'{output}'\nSTDERR:\n'{error}'";
-
-                    tcs.TrySetException(new InvalidOperationException(message));
-                }
-            };
+            var process = Process.Start(psi);
+            var result = stdout ? process.StandardOutput.ReadToEnd() : null;
```
Comment (OsmondJiang) [inline src/docfx/lib/ProcessUtility.cs]: will this meet out of buffer issue?

Comment (yufeih) [reply src/docfx/lib/ProcessUtility.cs]: I think it shouldn't hang as long as `ReadToEnd` is called before `WaitForExit` and we only read `stdout`. Test it with `git log` on azure-docs, it completed successfully.

File: src/docfx/lib/git/GitUtility.cs (hunk 1 above)
Comment (OsmondJiang) [reply src/docfx/lib/git/GitUtility.cs]: what about the log info for debugging when the execution failed?

Comment (yufeih) [reply src/docfx/lib/git/GitUtility.cs]: `stderr` will be piped to output, so you can see that from the log

**Pull request merged**